### PR TITLE
Convert to use Pipenv only

### PR DESCRIPTION
**Pull request by bseb**
### Background
Remove virtualenv and only use Pipenv to simplify workflow

### Changes
Update Pipfile, CI config and Makefile to use Pipenv,  remove requirements.txt

### Testing

* Tested in internal repo, working as expected.

Note: Once merged I will update confluence and Gitbook docs where appropriate

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,67 +7,60 @@ jobs:
...
-            version: v1.17.0-RC
-        - run:
-            name: Run schema tests
-            command: ./pantherlog -v && ./pantherlog test ./schemas || true
+          name: upload to hosted security
+          command: |
+            AWS_ACCESS_KEY_ID=$HOSTED_ACCESS_KEY_ID \
+            AWS_SECRET_ACCESS_KEY=$HOSTED_SECRET_ACCESS_KEY \
+            AWS_DEFAULT_REGION=$HOSTED_DEFAULT_REGION \
+            pipenv run -- panther_analysis_tool upload --filter Tags=hosted --aws-profile hosted
```
Comment (wey-chiang) [inline .circleci/config.yml:55]: ❓

I think this was meant for the dogfood repo

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,15 +1,4 @@
...
-commands:
-  download_pantherlog_tool:
-    parameters:
-      version:
-        type: string
-        description: The version tag to download
-    steps:
-        - run:
-            name: Download pantherlog tool
-            command: curl -sSO "https://panther-community-us-east-1.s3.amazonaws.com/<<parameters.version>>/tools/dev/pantherlog-linux-amd64" && mv pantherlog-linux-amd64 pantherlog && chmod +x pantherlog
```
Comment (wey-chiang) [inline .circleci/config.yml:12]: Are we removing pantherlog from the tests altogether?

**Review by wey-chiang — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (bseb) [reply .circleci/config.yml:55]: Yeah this is my mistake. I made a bad assumption that the CI job was the same for both, I'll be committing an updated Pipfile shortly

**Review by wey-chiang — APPROVED** [review]
LGTM, make sure the docs are updated once this is merged

Also, I expect this PR to cause issues with PR Sync for the syncing to the dogfood repo. Feel free to reach out if you need help w/ resolving those merge conflicts!

**Pull request merged**